client/src/modules/shipment/create-shipment.js

```diff
@@ -0,0 +1,321 @@
+    vm.validate();
+  };
+
+  vm.getLotByBarcode = function getLotByBarcode() {
+    Barcode.modal({ shouldSearch : false })
+      .then(record => {
+        if (record.uuid) {
+          vm.stockForm.addLotByBarcode(record.uuid);
+          vm.messages = vm.stockForm.messages();
+        }
+      });
+  };
+
+  vm.clear = function clear() {
+    vm.stockForm.clear();
+    // trigger on change depot behavior
+    // for having exit type and updated quantity
+    $timeout(() => {
+      onChangeDepot(vm.depot);
+    }, 0);
+  };
+
+  function onChangeDepot(depot) {
+    // depot assignment
+    vm.depot = depot;
+    vm.stockForm.setDepot(depot);
+
+    // set the shipment origin
+    vm.shipment.origin_depot_uuid = vm.depot.uuid;
+
+    // trick an exit type which is required
+    vm.stockForm.setExitType('loss');
+    vm.stockForm.setLossDistribution();
+
+    // run validation
+    vm.validate();
+
+    // refresh quantity available
+    return refreshQuantityAvailable(vm.depot);
+  }
+
+  function setReady(uuid) {
+    return ShipmentModal.setReadyForShipmentModal(uuid)
+      .then(res => {
+        if (res) { $state.go('shipments'); }
+      });
+  }
+
+  function getOverview(uuid) {
+    return ShipmentModal.shipmentOverviewModal(uuid);
+  }
+
+  function reset(form) {
+    const _form = form;
+    _form.$setPristine();
+    _form.$setUntouched();
+    vm.stockForm.store.clear();
+  }
+
+  function onRegisterApi(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function startup() {
+    vm.loading = true;
+    vm.hasError = false;
+
+    vm.stockForm.setup();
+    vm.validate();
+
+    // load the shipment for update
+    loadShipment();
+  }
+
+  function refreshQuantityAvailable(depot, dateTo = new Date()) {
+    return fetchAffectedAssets()
+      .then(assets => {
+        vm.alreadyAffected = assets;
+
+        vm.stockForm._pool.available.data.forEach(lot => {
+          const affectedQuantity = getAffectedQuantity(lot.lot_uuid);
+          lot._quantity_available -= affectedQuantity;
+          if (lot._quantity_available === 0 && affectedQuantity !== 0) {
+            vm.stockForm._pool.use(lot.lot_uuid);
+          }
+        });
+
+        vm.validate();
+
+        return Shipment.read(null, { depot_uuid : depot.uuid, dateTo, skipTags : true });
+      })
+      .then(list => {
+        vm.existingShipments = list;
+      });
+  }
+
+  function getAffectedQuantity(uuid) {
+    const el = (vm.alreadyAffected || []).filter(item => item.lot_uuid === uuid);
+    const quantity = el.reduce((prev, curr) => curr.quantity_sent + prev, 0);
+    return el ? quantity : 0;
+  }
+
+  function submit(form) {
+    if (form.$invalid) { return null; }
+
+    const isValidForSubmission = vm.stockForm.validate();
+
+    if (!isValidForSubmission) { return null; }
+
+    vm.$loading = true;
+    vm.shipment = cleanShipment(vm.shipment);
+    vm.shipment.lots = vm.stockForm.store.data.map(cleanShipmentItem);
+
+    const promise = (vm.isCreateState)
```

Comment:
Note: The Javascript for the ternary (conditional) expression here is correct, but generally the condition part is only in parenthesis if an expression, not a singly truthy variable.